arc_cache.go

```diff
@@ -250,16 +371,10 @@ func (b *arccache) cacheSize(c cid.Cid, blockSize int) {
 //
 // When ok is true, exists carries the correct answer, and size carries the
 // size, if known, or -1 if not.
-func (b *arccache) queryCache(k cid.Cid) (exists bool, size int, ok bool) {
+func (b *arccache) queryCache(k string) (exists bool, size int, ok bool) {
 	b.total.Inc()
-	if !k.Defined() {
-		log.Error("undefined cid in arccache")
-		// Return cache invalid so the call to blockstore happens
-		// in case of invalid key and correct error is created.
-		return false, -1, false
-	}
```

Comment:
We're removing this because the cache key is now pre-computed and passed into this function so we can't call `cid.Defined()`. We now check if it's defined at every call site before computing the cache key.